client/src/components/Tea/TeaList/TeaListContainer.jsx

```diff
@@ -4,23 +4,80 @@ import { deleteTea, getTeas } from "../../../actions/teaActions";
 import TeaList from "./TeaList";
 
 export class TeaListContainer extends React.Component {
+  state = {
+    sortedIDs: [],
+    memoizedIDs: {},
+    sortColumn: "",
+    sortOrder: ""
+  };
+
   handleDeleteClick = tea => {
     this.props.handleDelete(tea);
   };
 
+  handleSortClick = (columnName, sortOrder) => {
+    const list = this.props.teas.teaIDs;
+    let currentState = this.state.memoizedIDs;
+    let newSortOrder;
+
+    if (this.state.memoizedIDs[columnName]) {
+      newSortOrder = this.state.memoizedIDs[columnName][sortOrder];
+    } else {
+      newSortOrder = list.sort((a, b) => {
+        if (
+          this.props.teas.allTeas[a][columnName] >
+          this.props.teas.allTeas[b][columnName]
+        ) {
+          return 1;
+        }
+        if (
+          this.props.teas.allTeas[a][columnName] <
+          this.props.teas.allTeas[b][columnName]
+        ) {
+          return -1;
+        }
+        return 0;
+      });
+
+      const revSortOrder = [...newSortOrder].reverse();
+
+      currentState[columnName] = { asc: newSortOrder, desc: revSortOrder };
+    }
+    this.setState({
+      ...this.state,
+      sortedIDs: newSortOrder,
+      memoizedIDs: currentState,
+      sortColumn: columnName,
+      sortOrder: sortOrder
+    });
+  };
+
   componentDidMount() {
     this.props.getTeaList(this.props.userID);
   }
 
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.teas.teaIDs &&
+      this.props.teas.teaIDs !== prevProps.teas.teaIDs
+    ) {
+      this.setState({ sortedIDs: this.props.teas.teaIDs, memoizedIDs: {} });
```

Comment:
It looks like this will cause `sortedIDs` to reset to default, but the previous `sortColumn` and `sortOrder` will still be kept (as `setState` only shallowly changes the state - does not replace it entirely), so you'll have sort indicators on your table headers that don't match up with the data.